h2o-r/h2o-package/R/targetencoder.R

```diff
@@ -0,0 +1,136 @@
+#
+# -------------------------- Target Encoding  -------------------------- #
+#
+#' Create Target Encoding Map
+#'
+#' This is an API for a new target encoding implemented in JAVA.
+#'
+#' Creates a target encoding map based on group-by columns (`x`) and binary target column (`y`).
+#' Computing target encoding for high cardinality categorical columns can improve performance of supervised
+#' learning models.
+#'
+#' @param frame An H2OFrame object with which to create the target encoding map.
+#' @param x List of categorical column names or indices that we want apply target encoding to. 
```

Comment:
Thank you @jangorecki for bringing this up. Please take a look at a previous R's implementation. You can find it in frame.R file `h2o.target_encode_create`. In the comment it says that it is a list and that the items in the list can be a vectors of multiple columns. But in this PR's version... vector would be enough bc we don't support multicolumn items for now. @ledell should we change type of x to vector? I just don't know how critical it is for R's users.  I was trying to keep new API as close as possible to a previous one.

Reply:
The common reason for confusion is that python list type corresponds to R vector, not R list type. Vector should be sufficient to handle multiple columns, and list should be necessary when multiple sets of multiple columns are required, like `list("one_col_only", c("first_of_two", "second_of_two"), "one_col_again")`

Reply:
I think that a vector is the right choice for now.  Since we are already losing some support (for multiple column mappings), it won't be fully backwards compatible anyway.  We could also add a line of code that would support a one-element list in the old format, `list(c("A", "B"))`, and convert it into `c("A", "B")`... If we plan to support lists in the future,`list("one_col_only", c("first_of_two", "second_of_two"), "one_col_again")`, (in addition to a vector), that would be okay to add support for now. 

@jangorecki @deil87 What do you think?

Reply:
If multiple mappings may happen in near future then it is definitely better to use list. Ideally allowing to pass a vector which then will be `if (!is.list(x)) x = list(x)`, for convince. Any way you will choose won't introduce breaking changes so it can be easily changed later on.

Reply:
@ledell I'm a little bit confused. Let's define exactly what we will do.
I think that we will want to add multicolumn encodings in the near future. So I need to only add support for  "a one-element list in the old format, list(c("A", "B"))", right? Or anything else?

Reply:
@jangorecki Ok, thanks for clarifying.  I thought you were saying the opposite (that we should force the user to use a list for single column).

@deil87 Ok, so for now, let's go with a vector.  In the R client, once the user has passed in the vector, we can turn it into a list using `if (!is.list(x)) x = list(x)` before sending it to the H2O backend.  Does that make sense?

Reply:
@ledell backend expects vector. I also wanted to mention one more time that for now I actually don't know how we will support on backend side list of vectors parameter in the future. But I think we will be able to find a way. So ok, I will change type of parameter in R API  into vector type.